src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -13,16 +13,40 @@
 // limitations under the License.
 
 #include "ray/core_worker/transport/direct_task_transport.h"
+#include <ray/gcs/gcs_client/service_based_gcs_client.h>
 
 #include "ray/core_worker/transport/dependency_resolver.h"
-#include "ray/core_worker/transport/direct_actor_transport.h"
 
 namespace ray {
 
 Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
   RAY_LOG(DEBUG) << "Submit task " << task_spec.TaskId();
   resolver_.ResolveDependencies(task_spec, [this, task_spec]() {
     RAY_LOG(DEBUG) << "Task dependencies resolved " << task_spec.TaskId();
+    if (gcs_actor_management_enabled_ && task_spec.IsActorCreationTask()) {
+      RAY_CHECK(gcs_client_ != nullptr);
+      // If gcs actor management is enabled, the actor creation task will be send to gcs
+      // server directely after resolved the in-memory deptended objects.
+      // For more details please see the protocol of actor management based on gcs.
+      // https://docs.google.com/document/d/1EAWide-jy05akJp6OMtDn58XOK7bUyruWMia4E-fV28/edit?usp=sharing
+      auto actor_id = task_spec.ActorCreationId();
+      rpc::CreateActorRequest request;
+      request.mutable_task_spec()->CopyFrom(task_spec.GetMessage());
+      auto service_based_client =
+          std::dynamic_pointer_cast<gcs::ServiceBasedGcsClient>(gcs_client_);
+      auto &gcs_rpc_client = service_based_client->GetGcsRpcClient();
+      RAY_LOG(INFO) << "Creating actor " << actor_id;
+      gcs_rpc_client.CreateActor(
+          request, [actor_id](const Status &status, const rpc::CreateActorReply &reply) {
+            if (status.ok()) {
+              RAY_LOG(INFO) << "Succeed in creating actor " << actor_id;
+            } else {
+              RAY_LOG(FATAL) << "Failed to create actor " << actor_id;
```

Comment:
Thanks, it a good point!
Maybe the GCS Server is restarting and this should not affect the normal logic of the worker. It should retry again and again until the GCS Server reply something.
I will fix it.

Reply:
> We should not exit just because one actor failed to be created.

The gcs_rpc_client will ensure the message is send to the gcs server, if the driver/worker still receive a bad status, then there must be something wrong with the protocol of gcs-based actor management. So It's ok to exit here!

Reply:
What would happen if the server crashes? what's the detailed protocol for client to handle server's failure and restart?